server/src/main/java/org/spine3/server/entity/DefaultEntityStorageConverter.java

```diff
@@ -83,8 +85,13 @@ protected E doBackward(Tuple<I> tuple) {
 
         final EntityRecord record = tuple.getState();
         if (entity != null) {
-            entity.setState(state, record.getVersion());
-            entity.setVisibility(record.getVisibility());
+            if (entity instanceof AbstractVersionableEntity) {
+                AbstractVersionableEntity versionable = (AbstractVersionableEntity) entity;
```

Comment:
Should probably be declared `final`.

BTW, looks like an IDEA `instanceof` template should be edited :)